src/main/groovy/org/pih/warehouse/outbound/ImportPackingListItem.groovy

```diff
@@ -0,0 +1,98 @@
+package org.pih.warehouse.outbound
+
+import grails.databinding.BindUsing
+import grails.util.Holders
+import org.pih.warehouse.auth.AuthService
+import org.pih.warehouse.core.Constants
+import org.pih.warehouse.core.Location
+import org.pih.warehouse.core.Person
+import grails.validation.Validateable
+import org.pih.warehouse.inventory.InventoryItem
+import org.pih.warehouse.inventory.ProductAvailabilityService
+import org.pih.warehouse.product.Product
+
+
+class ImportPackingListItem implements Validateable {
+
+    String rowId
+
+    String palletName
+
+    String boxName
+
+    @BindUsing({ obj, source -> Product.findByProductCode(source['product']) })
+    Product product
+
+    String lotNumber
+
+    Date expirationDate
+
+    @BindUsing({ obj, source ->
+        Location location = Location.findByNameOrDefault(source['binLocation'])
+        // If location is not found, but we provided bin location name, it means, the location was not found,
+        // and we want to indicate that, instead of falling back to default (null)
+        // without this, we would then search e.g. for quantity available to promise for a product in the default bin location
+        if (!location && source['binLocation'] && !source['binLocation'].equalsIgnoreCase(Constants.DEFAULT_BIN_LOCATION_NAME)) {
+            // We want to indicate, that a bin location for given name has not been found.
+            obj.binLocationFound = false
```

Comment:
wouldn't it be better to move it to something like `EmptyStringsToNullBinder.groovy`? I would like to avoid adding a lot of code between properties, seeing this I am expecting to see methods under this, not the rest of the properties. But it can be just my opinion.

Reply:
yeah, this could be moved to a method, for now I've just focused on the functionality, but when I will do the final cleanup, I will take this suggestion into account.